### DISCUSSION TITLE
Is there a way to omit a step?

### DISCUSSION THREAD

**Discussion by yaotzin01** [allure-framework/allure2#2735] [Questions & Support] [answered]
I'm using Selenium with Cucumber, and Allure is mapping beforeStep and other "hooks" steps to report, and I don't want them there. I found a way to edit *-results.json file, but it works only when uploading results manually from IDE. When running from Bamboo, tests are going to Allure TestOps live, so editing file at the end doesn't work.
Is there a way to remove steps from report during execution?
I'm using Selenium Java (4.24.0), Cucumber (7.18.1), TestNG (7.10.2) and allure with testng and cucumber7 (2.29.0).
![attachment](https://github.com/user-attachments/assets/7a27dad3-322e-4d49-976d-3e2b910d5e67)

**Comment by baev** [comment]
Please specify the language, framework versions, and allure integration you're using

**Reply by yaotzin01** [reply]
done

**Comment by baev** [comment, marked as answer]
The easiest way would be to use the Allure Lifecycle Listener and remove some unnecessary steps [before the result is written to the file system](https://github.com/allure-framework/allure-java/blob/main/allure-java-commons/src/main/java/io/qameta/allure/listener/TestLifecycleListener.java#L59)

**Reply by yaotzin01** [reply]
That is what I was trying to do. I have a class that implements StepLifecycleListener and TestLifecycleListener, but I have no idea HOW to remove steps using it.

**Comment by yaotzin01** [comment]
Omg, I think I managed to do it.
One more question. How can I get rid of Set up and Tear down too?
![obraz](https://github.com/user-attachments/assets/9e0c1aa6-1cb1-40c3-ae81-b600155a1de8)

**Reply by baev** [reply]
https://github.com/allure-framework/allure-java/blob/main/allure-java-commons/src/main/java/io/qameta/allure/listener/ContainerLifecycleListener.java#L51

You can remove `befores`/`afters` items from the container or clear the `children` field to disable the container completely.

**Reply by yaotzin01** [reply]
Nice, it worked! Thanks!

**Reply by coolingsky** [reply]
@yaotzin01 May I know how you did that?

**Reply by yaotzin01** [reply]
@coolingsky I just remove steps in beforeTestWrite and beforeContainerWrite:
```
public class AllureListener implements TestLifecycleListener, ContainerLifecycleListener {

private boolean checkIfResultContainsServiceHooks(String resultName){
        return resultName.contains("ServiceHooks");
    }

    @Override
    public void beforeTestWrite(final TestResult result) {
        if (result.getStatus().equals(Status.BROKEN))
            result.setStatus(Status.FAILED);

        List<StepResult> steps = result.getSteps();
        for (int i = steps.size() - 1; i >= 0; i--)
            if (checkIfResultContainsServiceHooks(steps.get(i).getName()))
                steps.remove(i);
    }

@Override
    public void beforeContainerWrite(TestResultContainer container) {
        List<FixtureResult> beforeSteps = container.getBefores();
        for (int i = beforeSteps.size() - 1; i >= 0; i--)
            if (checkIfResultContainsServiceHooks(beforeSteps.get(i).getName()))
                beforeSteps.remove(i);

        List<FixtureResult> afterSteps = container.getAfters();
        for (int i = afterSteps.size() - 1; i >= 0; i--)
            if (checkIfResultContainsServiceHooks(afterSteps.get(i).getName()))
                afterSteps.remove(i);
    }
}
```

**Reply by coolingsky** [reply]
@yaotzin01 Great! It works perfectly! Thanks
